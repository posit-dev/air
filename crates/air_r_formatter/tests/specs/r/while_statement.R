while(a){}

while(a) {
  1 + 1
}

while({ complex }) {
  1 + 1
}

while(super_long_function_name_is_true_man_this_is_a_really_really_long_function()) {
  1 + 1
}

# ------------------------------------------------------------------------------
# Autobracing

while(a)a

while(a) 
  a

# ------------------------------------------------------------------------------
# Comments

while # leads while
(a) {
  b
}

while
# leads while
(a) {
  b
}

while(a # leads while
) {
  b
}

while(a
# leads while
) {
  b
}

while( # leads while
  a) {
  b
}

while(
  # leads while
  a) {
  b
}

while(
  a
  # leads while
) {
  b
}

while(a) # leads b
{
  b
}

while(a)
# leads b
{
  b
}

while(a) # dangles {}
{}

while(a) # leads b
b

while(a) # dangles {}
{
  # dangles {} 2
}
