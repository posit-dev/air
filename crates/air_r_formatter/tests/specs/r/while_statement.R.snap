---
source: crates/air_formatter_test/src/snapshot_builder.rs
info: r/while_statement.R
---
# Input

```R
while(a)a

while(a){}

while(a) {
  1 + 1
}

# TODO: Not entirely sure how this should be formatted.
# It's not very common though.
while({ complex }) {
  1 + 1
}

while(super_long_function_name_is_true_man_this_is_a_really_really_long_function()) {
  1 + 1
}

while(
  # comment
  a) {
  1 + 1
}

while(a
# comment
) {
  1 + 1
}

while(a # comment
) {
  1 + 1
}

while(a) # comment
{}

while(a) # comment
1

while(a) # comment1
{
  # comment2
}

```


=============================

# Outputs

## Output 1

-----
Indent style: Space
Indent width: 2
Line ending: LF
Line width: 80
Persistent line breaks: Respect
-----

```R
while (a) a

while (a) {
}

while (a) {
  1 + 1
}

# TODO: Not entirely sure how this should be formatted.
# It's not very common though.
while (
  {
    complex
  }
) {
  1 + 1
}

while (
  super_long_function_name_is_true_man_this_is_a_really_really_long_function()
) {
  1 + 1
}

while (
  # comment
  a
) {
  1 + 1
}

while (
  a
  # comment
) {
  1 + 1
}

while (
  a # comment
) {
  1 + 1
}

while (a) {
  # comment
}

while (a)
  # comment
  1

while (a) {
  # comment1
  # comment2
}
```
